includes/abilities/class-wp-core-abilities.php

```diff
@@ -0,0 +1,317 @@
+				'output_schema'       => array(
+					'type'                 => 'object',
+					'required'             => array( 'id', 'display_name', 'user_nicename', 'user_login', 'roles', 'locale' ),
+					'properties'           => array(
+						'id'            => array(
+							'type'        => 'integer',
+							'description' => __( 'The user ID.' ),
+						),
+						'display_name'  => array(
+							'type'        => 'string',
+							'description' => __( 'The display name of the user.' ),
+						),
+						'user_nicename' => array(
+							'type'        => 'string',
+							'description' => __( 'The URL-friendly name for the user.' ),
+						),
+						'user_login'    => array(
+							'type'        => 'string',
+							'description' => __( 'The login username for the user.' ),
+						),
+						'roles'         => array(
+							'type'        => 'array',
+							'description' => __( 'The roles assigned to the user.' ),
+							'items'       => array(
+								'type' => 'string',
+							),
+						),
+						'locale'        => array(
+							'type'        => 'string',
+							'description' => __( 'The locale string for the user, such as en_US.' ),
+						),
+					),
+					'additionalProperties' => false,
+				),
+				'execute_callback'    => static function (): array {
+					$current_user = wp_get_current_user();
+
+					return array(
+						'id'            => $current_user->ID,
+						'display_name'  => $current_user->display_name,
+						'user_nicename' => $current_user->user_nicename,
+						'user_login'    => $current_user->user_login,
+						'roles'         => $current_user->roles,
+						'locale'        => get_user_locale( $current_user ),
+					);
+				},
+				'permission_callback' => static function (): bool {
+					return is_user_logged_in();
+				},
+				'meta'                => array(
+					'annotations'  => array(
+						'readonly'    => true,
+						'destructive' => false,
+						'idempotent'  => true,
+					),
+					'show_in_rest' => false,
+				),
+			)
+		);
+	}
+
+	/**
+	 * Registers the `core/get-environment-info` ability.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return void
+	 */
+	protected static function register_get_environment_info(): void {
+		wp_register_ability(
+			'core/get-environment-info',
+			array(
+				'label'               => __( 'Get Environment Info' ),
+				'description'         => __( 'Returns core details about the site\'s runtime context for diagnostics and compatibility (environment, PHP runtime, database server info, WordPress version).' ),
+				'category'            => self::CATEGORY_SITE,
+				'output_schema'       => array(
+					'type'                 => 'object',
+					'required'             => array( 'environment', 'php_version', 'db_server_info', 'wp_version' ),
+					'properties'           => array(
+						'environment'    => array(
+							'type'        => 'string',
+							'description' => __( 'The site\'s runtime environment classification (e.g., production, staging, development).' ),
```

Comment:
This can only be one of four values.  I think it's valueable to make that clearer.